Lib/test/librptest/rptest.py

```diff
@@ -0,0 +1,51 @@
+
+
+from .imported import original
+from .ours import *
+from .core import *
+
+import unittest
+import types
+import re
+
+# regex to parse urls
+url_regex = re.compile(
+        r'^(?:http|ftp)s?://' # http:// or https://
+        r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|' #domain...
+        r'localhost|' #localhost...
+        r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})' # ...or ip
+        r'(?::\d+)?' # optional port
+        r'(?:/?|[/?]\S+)$', re.IGNORECASE)
+
+
+def originated_from(major_version, minor_version, original_file_link):
```

Comment:
Maybe this could take a patch version as well?

Reply:
The information here relates to the Python language version, the file itself can be from a micro CPython version. This is covered by the link. But the relevant information for the compatibility if it is 3.7 or 3.8 or 3.9.